### PR TITLE
NFC - minor spelling tweaks under compiler directory 

### PR DESCRIPTION
**Pull request by kiszk**
This PR addresses minor spelling tweaks under `tensorflow/compiler` directory. In addition, `RELEASE.MD` is also addressed.

### CODE REVIEW

**Comment by kiszk** [comment]
Would it be possible to let me know if smaller PRs (e.g. 100 files per PR) are better for ease of review? I will split this into two PRs.

File: tensorflow/compiler/xla/service/hlo_computation.h [hunk 1]
```diff
@@ -419,7 +419,7 @@ class HloComputation {
   // the HLO computation with the exception of fusion computation. A parameter
   // instruction is removable for a fusion computation.
   //
-  // Note that IsSafelyRemovable() is a necassarily condition to remove an
+  // Note that IsSafelyRemovable() is a necessarily condition to remove an
```
Comment (mihaimaruseac) [inline tensorflow/compiler/xla/service/hlo_computation.h]: `necessary`?

**Review by mihaimaruseac — DISMISSED** [review]
That is great! Thank you for the PR.

Did you use a script to do the fixing? If yes, can you please share it too?

File: tensorflow/compiler/xla/service/hlo_computation.h (hunk 1 above)
Comment (kiszk) [reply tensorflow/compiler/xla/service/hlo_computation.h]: Good catch, done!

**Review by mihaimaruseac — APPROVED** [review]

**Comment by kiszk** [comment]
I use a script to list up potential candidates. The script lists up a lot of false positives. Finally, I manually find and fix them.
Does my script help you even if it looks premature?

An example of outputs
```
...
abstrat
absw
absx
abt
abuf
abx
abz
aca
acad
acat
acb
acbb
acbd
acc
accbuffer
acccbbdddd
accd
acce
accel
...
```

**Comment by mihaimaruseac** [comment]
We might need to use the script internally to convert tooling that expects some of the same typos to exist.

**Comment by kiszk** [comment]
I use this script from https://qiita.com/debility/items/d3b24bc9a941241e82b8
```
find . -type f -print |xargs file |grep text |awk -F' ' '{print $1}' |sed -e 's/://g' |xargs cat |sed "s/[A-Z][a-z]/ \0/g" |tr '[A-Z]' '[a-z]' |aspell list |sort|uniq
```

**Comment by kiszk** [comment]
I am working for other directories now.

**Pull request merged**